src/main/java/mate/jdbc/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.dao.DataProcessingException;
+import mate.jdbc.dao.ManufacturerDao;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers (name, country) VALUES (?,?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createStatement
+                        = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            createStatement.setString(1, manufacturer.getName());
+            createStatement.setString(2, manufacturer.getCountry());
+            createStatement.executeUpdate();
+            ResultSet generatedKeys = createStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't insert manufacturer " + manufacturer,
+                    throwables);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        Optional<Manufacturer> manufacturer = null;
+        String query = "SELECT * FROM manufacturers WHERE is_deleted = FALSE AND id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getStatement = connection.prepareStatement(query)) {
+            getStatement.setObject(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturer = Optional.ofNullable(retrieveManufacturerFromResultSet(resultSet));
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't get manufacturer by id " + id, throwables);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        String query = "SELECT * FROM manufacturers WHERE is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement = connection.prepareStatement(query)) {
```

Comment:
```suggestion
                PreparedStatement statement = connection.prepareStatement(query)) {
```
We already see the method name, you can just call it `statement`